src/se/vidstige/jadb/managers/ActivityManager.java

```diff
@@ -0,0 +1,41 @@
+package se.vidstige.jadb.managers;
+
+import se.vidstige.jadb.JadbDevice;
+import se.vidstige.jadb.JadbException;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Author: pablobaxter.
+ */
+public class ActivityManager {
+
+    private final JadbDevice device;
+
+    public ActivityManager(JadbDevice device) {
+        this.device = device;
+    }
+
+    public void broadcastIntent(Intent intent, String permission) throws IOException, JadbException {
+        List<String> args = new ArrayList<>();
+        args.add("broadcast");
+        intent.generate(args);
+        if(permission != null){
+            args.add("--receiver-permission");
+            args.add(permission);
+        }
+        InputStream stream = device.executeShell("am", args.toArray(new String[args.size()]));
+        int i;
+        do{
+            i = stream.read();
+        } while (i != -1);
```

Comment:
Why are you reading the entire stream and then closing? Can we just close here, or do we need to read the entire stream? In any case, move it out as a function. Perhaps a `Stream` class with a static method `skipAll` or something.